File: 18-9-2024/Flip the cards.py
# Author: Tajwar Saiyeed
# Date: 2024-09-18 07:21:00
# File: Flip the cards.cpp

tc = int(input())
for _ in range(tc):
    n, x = map(int, input().split())
    if n == x:
        print(0)
    else: 
        print(min(n-x, x))

# Problem : https://www.codechef.com/practice/course/logical-problems/DIFF800/problems/FLIPCARDS

File: 18-9-2024/Flip the cards.cpp
/*
    Author: Tajwar Saiyeed
    Date: 2024-09-18 07:20:24
    File: Flip the cards.cpp
*/

#include <bits/stdc++.h>
using namespace std;

int main()
{
    int tc, x, n;
    cin >> tc;
    while (tc--)
    {
        cin >> n >> x;
        if (n == x)
            cout << 0 << '\n';
        else
            cout << min(n - x, x) << '\n';
    }
    return 0;
}

// Problem : https://www.codechef.com/practice/course/logical-problems/DIFF800/problems/FLIPCARDS